src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.TransactionExecutor.cs

```diff
@@ -43,9 +44,28 @@ protected override ResultWrapper<TResult> Execute(BlockHeader header, Transactio
                 return ExecuteTx(clonedHeader, tx, token);
             }
 
-            private static bool ShouldSetBaseFee(TransactionForRpc t) =>
-                // x?.IsZero == false <=> x > 0
-                t.GasPrice?.IsZero == false || t.MaxFeePerGas?.IsZero == false || t.MaxPriorityFeePerGas?.IsZero == false;
+            // TODO: Should we move this method to `TransactionForRpc` directly?
+            private static bool ShouldSetBaseFee(TransactionForRpc t)
+            {
+                var positiveGasPrice = false;
+                if (t is LegacyTransactionForRpc legacy)
+                {
+                    positiveGasPrice = IsPositive(legacy.GasPrice);
+                }
+
+                var positiveMaxFeePerGas = false;
+                var positiveMaxPriorityFeePerGas = false;
+                if (t is EIP1559TransactionForRpc eip1559)
+                {
+                    positiveMaxFeePerGas = IsPositive(eip1559.MaxFeePerGas);
+                    positiveMaxPriorityFeePerGas = IsPositive(eip1559.MaxPriorityFeePerGas);
+                }
+
+                return positiveGasPrice || positiveMaxFeePerGas || positiveMaxPriorityFeePerGas;
+
+                // value?.IsZero == false <=> x > 0
+                static bool IsPositive(UInt256? value) => value?.IsZero == false;
+            }
```

Comment:
As the comment says, should we move this method to the base `TransactionForRpc` class directly?

Reply:
Moved to the base `TransactionForRpc` class